Zyppng

jreidinger opened the pull request:
## Problem

*Short description of the original problem.*

- https://trello.com/c/USzirIEA

## Solution

*Short description of the fix.*


## Testing

- *Added a new unit test*
- *Tested manually*


## Screenshots
Example output from my distrobox:
```
System repos:
Some("openSUSE-Tumbleweed-Debug")
Some("openSUSE-Tumbleweed-Non-Oss")
Some("Open H.264 Codec (openSUSE Tumbleweed)")
Some("openSUSE-Tumbleweed-Oss")
Some("openSUSE-Tumbleweed-Source")
Some("openSUSE-Tumbleweed-Update")

Host repos:
Some("repo-non-free (20240910)")
Some("The Latest YaST Packages")
Some("Visual Studio Code")
Some("Rust (15.5)")
Some("nVidia Graphics Drivers")
Some("fedora40")
Some("google-chrome")
Some("X11:Utilities")
Some("repo-non-oss (20240910)")
Some("repo-non-oss-debug (20240910)")
Some("repo-openh264 (20240910)")
Some("repo-oss (20240910)")
Some("repo-oss-debug (20240910)")
Some("repo-oss-source (20240910)")
Some("update-backports (20240910)")
Some("update-backports-debug (20240910)")
Some("update-non-oss (20240910)")
Some("update-non-oss-debug (20240910)")
Some("update-oss (20240910)")
Some("update-oss-debug (20240910)")
Some("update-sle (20240910)")
Some("update-sle-debug (20240910)")
Some("Libdvdcss Repository")
Some("packman")
Some("Source Repository (Non-OSS)")

```

jreidinger committed:
initial bindings for zyppng

jreidinger committed:
fix build of Context

jreidinger committed:
add exceptions with TODO for init macro

jreidinger committed:
add zypp expected

jreidinger committed:
fix expected generation

jreidinger committed:
add zyp info base

jreidinger committed:
generate rest of current structs

jreidinger committed:
generate also builders

jreidinger committed:
add example binary

jreidinger committed:
improve example

mvidner commented:
FWIW, when I build this branch in the `/testing_using_container.sh` setup, 
1. Compilation fails unless I `zypper install glib2-devel`
2. Then it fails because it cannot link `-lzypp-glib` ... so I will need to build a modified libzypp for that.

jreidinger committed:
first version of setup script and try to improve submodules

jreidinger committed:
more git module changes

jreidinger committed:
exclude gir from workspace

jreidinger committed:
fix rubygem name

jreidinger committed:
workaround issue with gir workspace

jreidinger committed:
make script idempotent

jreidinger committed:
fix build libzypp with glib

jreidinger committed:
update to the fixed version of libzypp

jreidinger committed:
document how to run script with distrobox

jreidinger committed:
Update gir-files location for zypp and simplify lib.rs

jreidinger committed:
autogenerate documentaion with help of rustdoc-stripper

jreidinger committed:
Add readme and install doc tool

mvidner commented:
Thanks for the readme and the docs!

Right now the PR is a Draft targeting `master`. Do you have a plan to make it mergeable soon? Or is that unknown and to be determined in the review meeting and/or with Zypp team?

jreidinger commented:
@mvidner plan is not to merge it now as it is build as part of cargo build in rust directory. So it is just draft to demonstrate it and start of discussion. Maybe we will need to hide it with some feature flag.

mvidner committed:
setup_env.sh: sudo only where it is available

in the Agama testing containers it is missing

mvidner committed:
setup_env.sh: parallel make (nproc)

mvidner committed:
setup_env.sh: remove rust RPMs before installing via rustup

the Agama testing container has them

mvidner committed:
setup_env.sh: save space by not building zypp autodocs

mvidner committed:
setup_env.sh: a readable error when sudo is missing

It still fails but explains why
Before, it just failed because of set -e and missing sudo

mvidner commented:
@jreidinger I have adjusted the setup script, to work in a `agama-testing:latest` podman container.

It still fails for me with distrobox + 2024-09-17 Tumbleweed (not root, sudo missing), I wonder why it works for you

mvidner committed:
setup_env.sh: make executable

mvidner commented:
BTW we may want to set up the new zypp in `/usr/local` because after it gets installed, it interferes with package management :rofl: 
```console
# zypper se nosuchpackage
[...]
No matching items found.
[...]
corrupted size vs. prev_size in fastbins
^C��
    &V^C�EFV^C�EFV^\Quit (core dumped)
```

It could be as easy as
`cmake   -D CMAKE_INSTALL_PREFIX=/usr/local ...`

mvidner committed:
testing_using_container: fix unbound variable

mvidner committed:
setup_env.sh: don't invoke cmake twice

Makefile.cvs would do it in build/ and then here in ./
Call it once, in build/

jreidinger commented:
lets close PR for now as project is send to ice
